inductiva/world/terrain.py

```diff
@@ -0,0 +1,175 @@
+"""Represent a complex terrain profile."""
+import typing
+import random
+
+import pyvista as pv
+
+import inductiva
+from inductiva.generative import procedural
+
+
+class Terrain:
+    """Represents a Terrain profile.
+    
+    A Terrain profile is defined by the surface of the terrain.
+    Here, we describe it by a mesh with coordinates (x, y, z).
+    The (x, y) reflect the (longitude, latitude) of the terrain
+    and the z coordinate is the elevation of the terrain.
+    """
+
+    def __init__(self, terrain_mesh):
+        """Initializes the `Terrain` object.
+        
+        Args:
+            terrain_mesh (pyvista.StructuredGrid): Terrain mesh."""
+        self.mesh = terrain_mesh
+
+    @classmethod
+    def from_file(cls, terrain_file: str):
+        """Creates a `Terrain` object from a text file."""
+        terrain_mesh = pv.read(terrain_file)
+
+        return cls(terrain_mesh)
+
+    @classmethod
+    def from_random_generation(cls,
+                               x_range: typing.Sequence[float],
+                               y_range: typing.Sequence[float],
+                               num_list: typing.List[int],
+                               initial_roughness: float = 1,
+                               roughness_factor: float = 0.5,
+                               min_elevation: float = 0,
+                               max_elevation: float = 1,
+                               random_seed: int = None):
+        """Creates a `Terrain` object with random elevations.
+        The elevation of the corners are randomly selected within
+        the range of 0 to 1. This limits the maximum terrain height
+        due to the nature of the terrain generation algorithm.
+        To allow for more complex terrains, we allow the user to multiply
+        by a `heigh_factor` to increase the maximum height of the terrain.
+        In this way, the maximum_height possible will be around 
+        the value of the `height_factor`.
+
+        TODO: Improve terrain generation for high discrepancy
+        terrains.
+
+        Args:
+            x_range = [x_min, x_max]: Range of the x-axis in meters (m).
+            y_range = [y_min, y_max]: Range of the y-axis in meters (m).
+            num_list: List of number of points in both directions
+                [x_num, y_num].
+            initial_roughness: Initial roughness of the terrain.
+            roughness_factor: Factor to multiply the roughness by.
+            min_elevation: Minimum elevation of the terrain.
+            max_elevation: Maximum elevation of the terrain.
+            random_seed: Random seed to use for the terrain generation.
+        """
+        random.seed(random_seed)
+        corner_values = [
```

Comment:
We want to be able to set each of these *individually* so we can have more control on the type of landscape.
For example: a plain would have these ranging in equal interavals, but a slope would have two of these high and two low.